### ISSUE TITLE
Plugin fails with Psalm 4 and PHP 8.2

### ISSUE THREAD

**Issue by rogervila** [psalm/psalm-vscode-plugin#214]
# Behaviour

## Expected

XXX

## Actual

XXX

## Steps to reproduce:

[**NOTE**: Self-contained, minimal reproducing code samples are **extremely** helpful and will expedite addressing your issue]

1. Open a PHP 8.2 project that has psalm 4
2. An error is triggered

# Diagnostic data

-   PHP version: PHP 8.2.0 (cli) (built: Dec  9 2022 16:19:06) (NTS)
Copyright (c) The PHP Group
Zend Engine v4.2.0, Copyright (c) Zend Technologies
    with Xdebug v3.2.0, Copyright (c) 2002-2022, by Derick Rethans
    with Zend OPcache v8.2.0, Copyright (c), by Zend Technologies

-   Psalm version: 4.30.0

<details>

<summary>"Psalm Language Server" channel in the OUTPUT panel (Last 1000 lines)</summary>

<p>

```
[INFO  - 11:36:05 AM] Starting language server
[INFO  - 11:36:06 AM] Psalm Version was detected as 4.30.0
[ERROR  - 11:36:06 AM] Xdebug: [Config] The setting 'xdebug.remote_autostart' has been renamed, see the upgrading guide at https://xdebug.org/docs/upgrade_guide#changed-xdebug.remote_autostart (See: https://xdebug.org/docs/errors#CFG-C-CHANGED)
Xdebug: [Config] The setting 'xdebug.remote_enable' has been renamed, see the upgrading guide at https://xdebug.org/docs/upgrade_guide#changed-xdebug.remote_enable (See: https://xdebug.org/docs/errors#CFG-C-CHANGED)

Xdebug: [Config] The setting 'xdebug.remote_autostart' has been renamed, see the upgrading guide at https://xdebug.org/docs/upgrade_guide#changed-xdebug.remote_autostart (See: https://xdebug.org/docs/errors#CFG-C-CHANGED)
Xdebug: [Config] The setting 'xdebug.remote_enable' has been renamed, see the upgrading guide at https://xdebug.org/docs/upgrade_guide#changed-xdebug.remote_enable (See: https://xdebug.org/docs/errors#CFG-C-CHANGED)

[ERROR  - 11:36:07 AM] Uncaught RuntimeException: PHP Error: Creation of dynamic property Psalm\Internal\LanguageServer\LanguageServer::$mapper is deprecated in /path/to/project/vendor/felixfbecker/advanced-json-rpc/lib/Dispatcher.php:53 in /path/to/project/vendor/vimeo/psalm/src/Psalm/Internal/ErrorHandler.php:66
Stack trace:
#0 /path/to/project/vendor/felixfbecker/advanced-json-rpc/lib/Dispatcher.php(53): Psalm\Internal\ErrorHandler::Psalm\Internal\{closure}(8192, 'Creation of dyn...', '/path/to...', 53)
#1 /path/to/project/vendor/vimeo/psalm/src/Psalm/Internal/LanguageServer/LanguageServer.php(116): AdvancedJsonRpc\Dispatcher->__construct(Object(Psalm\Internal\LanguageServer\LanguageServer), '/')
#2 /path/to/project/vendor/vimeo/psalm/src/Psalm/Internal/Analyzer/ProjectAnalyzer.php(555): Psalm\Internal\LanguageServer\LanguageServer->__construct(Object(Psalm\Internal\LanguageServer\ProtocolStreamReader), Object(Psalm\Internal\LanguageServer\ProtocolStreamWriter), Object(Psalm\Internal\Analyzer\ProjectAnalyzer))
#3 /path/to/project/vendor/vimeo/psalm/src/Psalm/Internal/Cli/LanguageServer.php(322): Psalm\Internal\Analyzer\ProjectAnalyzer->server(NULL, false)
#4 /path/to/project/vendor/vimeo/psalm/psalm-language-server(7): Psalm\Internal\Cli\LanguageServer::run(Array)
#5 {main}
(Psalm 4.30.0@d0bc6e25d89f649e4f36a534f330f8bb4643dd69 crashed due to an uncaught Throwable)

Uncaught RuntimeException: PHP Error: Creation of dynamic property Psalm\Internal\LanguageServer\LanguageServer::$mapper is deprecated in /path/to/project/vendor/felixfbecker/advanced-json-rpc/lib/Dispatcher.php:53 in /path/to/project/vendor/vimeo/psalm/src/Psalm/Internal/ErrorHandler.php:66
Stack trace:
#0 /path/to/project/vendor/felixfbecker/advanced-json-rpc/lib/Dispatcher.php(53): Psalm\Internal\ErrorHandler::Psalm\Internal\{closure}(8192, 'Creation of dyn...', '/path/to...', 53)
#1 /path/to/project/vendor/vimeo/psalm/src/Psalm/Internal/LanguageServer/LanguageServer.php(116): AdvancedJsonRpc\Dispatcher->__construct(Object(Psalm\Internal\LanguageServer\LanguageServer), '/')
#2 /path/to/project/vendor/vimeo/psalm/src/Psalm/Internal/Analyzer/ProjectAnalyzer.php(555): Psalm\Internal\LanguageServer\LanguageServer->__construct(Object(Psalm\Internal\LanguageServer\ProtocolStreamReader), Object(Psalm\Internal\LanguageServer\ProtocolStreamWriter), Object(Psalm\Internal\Analyzer\ProjectAnalyzer))
#3 /path/to/project/vendor/vimeo/psalm/src/Psalm/Internal/Cli/LanguageServer.php(322): Psalm\Internal\Analyzer\ProjectAnalyzer->server(NULL, false)
#4 /path/to/project/vendor/vimeo/psalm/psalm-language-server(7): Psalm\Internal\Cli\LanguageServer::run(Array)
#5 {main}
(Psalm 4.30.0@d0bc6e25d89f649e4f36a534f330f8bb4643dd69 crashed due to an uncaught Throwable)

[Info  - 11:36:07 AM] Connection to server got closed. Server will restart.
[INFO  - 11:36:07 AM] Psalm Version was detected as 4.30.0
[ERROR  - 11:36:07 AM] Xdebug: [Config] The setting 'xdebug.remote_autostart' has been renamed, see the upgrading guide at https://xdebug.org/docs/upgrade_guide#changed-xdebug.remote_autostart (See: https://xdebug.org/docs/errors#CFG-C-CHANGED)
Xdebug: [Config] The setting 'xdebug.remote_enable' has been renamed, see the upgrading guide at https://xdebug.org/docs/upgrade_guide#changed-xdebug.remote_enable (See: https://xdebug.org/docs/errors#CFG-C-CHANGED)

Xdebug: [Config] The setting 'xdebug.remote_autostart' has been renamed, see the upgrading guide at https://xdebug.org/docs/upgrade_guide#changed-xdebug.remote_autostart (See: https://xdebug.org/docs/errors#CFG-C-CHANGED)
Xdebug: [Config] The setting 'xdebug.remote_enable' has been renamed, see the upgrading guide at https://xdebug.org/docs/upgrade_guide#changed-xdebug.remote_enable (See: https://xdebug.org/docs/errors#CFG-C-CHANGED)

[ERROR  - 11:36:08 AM] Uncaught RuntimeException: PHP Error: Creation of dynamic property Psalm\Internal\LanguageServer\LanguageServer::$mapper is deprecated in /path/to/project/vendor/felixfbecker/advanced-json-rpc/lib/Dispatcher.php:53 in /path/to/project/vendor/vimeo/psalm/src/Psalm/Internal/ErrorHandler.php:66
Stack trace:
#0 /path/to/project/vendor/felixfbecker/advanced-json-rpc/lib/Dispatcher.php(53): Psalm\Internal\ErrorHandler::Psalm\Internal\{closure}(8192, 'Creation of dyn...', '/path/to...', 53)
#1 /path/to/project/vendor/vimeo/psalm/src/Psalm/Internal/LanguageServer/LanguageServer.php(116): AdvancedJsonRpc\Dispatcher->__construct(Object(Psalm\Internal\LanguageServer\LanguageServer), '/')
#2 /path/to/project/vendor/vimeo/psalm/src/Psalm/Internal/Analyzer/ProjectAnalyzer.php(555): Psalm\Internal\LanguageServer\LanguageServer->__construct(Object(Psalm\Internal\LanguageServer\ProtocolStreamReader), Object(Psalm\Internal\LanguageServer\ProtocolStreamWriter), Object(Psalm\Internal\Analyzer\ProjectAnalyzer))
#3 /path/to/project/vendor/vimeo/psalm/src/Psalm/Internal/Cli/LanguageServer.php(322): Psalm\Internal\Analyzer\ProjectAnalyzer->server(NULL, false)
#4 /path/to/project/vendor/vimeo/psalm/psalm-language-server(7): Psalm\Internal\Cli\LanguageServer::run(Array)
#5 {main}
(Psalm 4.30.0@d0bc6e25d89f649e4f36a534f330f8bb4643dd69 crashed due to an uncaught Throwable)

Uncaught RuntimeException: PHP Error: Creation of dynamic property Psalm\Internal\LanguageServer\LanguageServer::$mapper is deprecated in /path/to/project/vendor/felixfbecker/advanced-json-rpc/lib/Dispatcher.php:53 in /path/to/project/vendor/vimeo/psalm/src/Psalm/Internal/ErrorHandler.php:66
Stack trace:
#0 /path/to/project/vendor/felixfbecker/advanced-json-rpc/lib/Dispatcher.php(53): Psalm\Internal\ErrorHandler::Psalm\Internal\{closure}(8192, 'Creation of dyn...', '/path/to...', 53)
#1 /path/to/project/vendor/vimeo/psalm/src/Psalm/Internal/LanguageServer/LanguageServer.php(116): AdvancedJsonRpc\Dispatcher->__construct(Object(Psalm\Internal\LanguageServer\LanguageServer), '/')
#2 /path/to/project/vendor/vimeo/psalm/src/Psalm/Internal/Analyzer/ProjectAnalyzer.php(555): Psalm\Internal\LanguageServer\LanguageServer->__construct(Object(Psalm\Internal\LanguageServer\ProtocolStreamReader), Object(Psalm\Internal\LanguageServer\ProtocolStreamWriter), Object(Psalm\Internal\Analyzer\ProjectAnalyzer))
#3 /path/to/project/vendor/vimeo/psalm/src/Psalm/Internal/Cli/LanguageServer.php(322): Psalm\Internal\Analyzer\ProjectAnalyzer->server(NULL, false)
#4 /path/to/project/vendor/vimeo/psalm/psalm-language-server(7): Psalm\Internal\Cli\LanguageServer::run(Array)
#5 {main}
(Psalm 4.30.0@d0bc6e25d89f649e4f36a534f330f8bb4643dd69 crashed due to an uncaught Throwable)

[Info  - 11:36:08 AM] Connection to server got closed. Server will restart.
[INFO  - 11:36:08 AM] Psalm Version was detected as 4.30.0
[ERROR  - 11:36:09 AM] Xdebug: [Config] The setting 'xdebug.remote_autostart' has been renamed, see the upgrading guide at https://xdebug.org/docs/upgrade_guide#changed-xdebug.remote_autostart (See: https://xdebug.org/docs/errors#CFG-C-CHANGED)
Xdebug: [Config] The setting 'xdebug.remote_enable' has been renamed, see the upgrading guide at https://xdebug.org/docs/upgrade_guide#changed-xdebug.remote_enable (See: https://xdebug.org/docs/errors#CFG-C-CHANGED)

Xdebug: [Config] The setting 'xdebug.remote_autostart' has been renamed, see the upgrading guide at https://xdebug.org/docs/upgrade_guide#changed-xdebug.remote_autostart (See: https://xdebug.org/docs/errors#CFG-C-CHANGED)
Xdebug: [Config] The setting 'xdebug.remote_enable' has been renamed, see the upgrading guide at https://xdebug.org/docs/upgrade_guide#changed-xdebug.remote_enable (See: https://xdebug.org/docs/errors#CFG-C-CHANGED)

[ERROR  - 11:36:09 AM] Uncaught RuntimeException: PHP Error: Creation of dynamic property Psalm\Internal\LanguageServer\LanguageServer::$mapper is deprecated in /path/to/project/vendor/felixfbecker/advanced-json-rpc/lib/Dispatcher.php:53 in /path/to/project/vendor/vimeo/psalm/src/Psalm/Internal/ErrorHandler.php:66
Stack trace:
#0 /path/to/project/vendor/felixfbecker/advanced-json-rpc/lib/Dispatcher.php(53): Psalm\Internal\ErrorHandler::Psalm\Internal\{closure}(8192, 'Creation of dyn...', '/path/to...', 53)
#1 /path/to/project/vendor/vimeo/psalm/src/Psalm/Internal/LanguageServer/LanguageServer.php(116): AdvancedJsonRpc\Dispatcher->__construct(Object(Psalm\Internal\LanguageServer\LanguageServer), '/')
#2 /path/to/project/vendor/vimeo/psalm/src/Psalm/Internal/Analyzer/ProjectAnalyzer.php(555): Psalm\Internal\LanguageServer\LanguageServer->__construct(Object(Psalm\Internal\LanguageServer\ProtocolStreamReader), Object(Psalm\Internal\LanguageServer\ProtocolStreamWriter), Object(Psalm\Internal\Analyzer\ProjectAnalyzer))
#3 /path/to/project/vendor/vimeo/psalm/src/Psalm/Internal/Cli/LanguageServer.php(322): Psalm\Internal\Analyzer\ProjectAnalyzer->server(NULL, false)
#4 /path/to/project/vendor/vimeo/psalm/psalm-language-server(7): Psalm\Internal\Cli\LanguageServer::run(Array)
#5 {main}
(Psalm 4.30.0@d0bc6e25d89f649e4f36a534f330f8bb4643dd69 crashed due to an uncaught Throwable)

Uncaught RuntimeException: PHP Error: Creation of dynamic property Psalm\Internal\LanguageServer\LanguageServer::$mapper is deprecated in /path/to/project/vendor/felixfbecker/advanced-json-rpc/lib/Dispatcher.php:53 in /path/to/project/vendor/vimeo/psalm/src/Psalm/Internal/ErrorHandler.php:66
Stack trace:
#0 /path/to/project/vendor/felixfbecker/advanced-json-rpc/lib/Dispatcher.php(53): Psalm\Internal\ErrorHandler::Psalm\Internal\{closure}(8192, 'Creation of dyn...', '/path/to...', 53)
#1 /path/to/project/vendor/vimeo/psalm/src/Psalm/Internal/LanguageServer/LanguageServer.php(116): AdvancedJsonRpc\Dispatcher->__construct(Object(Psalm\Internal\LanguageServer\LanguageServer), '/')
#2 /path/to/project/vendor/vimeo/psalm/src/Psalm/Internal/Analyzer/ProjectAnalyzer.php(555): Psalm\Internal\LanguageServer\LanguageServer->__construct(Object(Psalm\Internal\LanguageServer\ProtocolStreamReader), Object(Psalm\Internal\LanguageServer\ProtocolStreamWriter), Object(Psalm\Internal\Analyzer\ProjectAnalyzer))
#3 /path/to/project/vendor/vimeo/psalm/src/Psalm/Internal/Cli/LanguageServer.php(322): Psalm\Internal\Analyzer\ProjectAnalyzer->server(NULL, false)
#4 /path/to/project/vendor/vimeo/psalm/psalm-language-server(7): Psalm\Internal\Cli\LanguageServer::run(Array)
#5 {main}
(Psalm 4.30.0@d0bc6e25d89f649e4f36a534f330f8bb4643dd69 crashed due to an uncaught Throwable)

[Info  - 11:36:09 AM] Connection to server got closed. Server will restart.
[INFO  - 11:36:09 AM] Psalm Version was detected as 4.30.0
[ERROR  - 11:36:10 AM] Xdebug: [Config] The setting 'xdebug.remote_autostart' has been renamed, see the upgrading guide at https://xdebug.org/docs/upgrade_guide#changed-xdebug.remote_autostart (See: https://xdebug.org/docs/errors#CFG-C-CHANGED)
Xdebug: [Config] The setting 'xdebug.remote_enable' has been renamed, see the upgrading guide at https://xdebug.org/docs/upgrade_guide#changed-xdebug.remote_enable (See: https://xdebug.org/docs/errors#CFG-C-CHANGED)

Xdebug: [Config] The setting 'xdebug.remote_autostart' has been renamed, see the upgrading guide at https://xdebug.org/docs/upgrade_guide#changed-xdebug.remote_autostart (See: https://xdebug.org/docs/errors#CFG-C-CHANGED)
Xdebug: [Config] The setting 'xdebug.remote_enable' has been renamed, see the upgrading guide at https://xdebug.org/docs/upgrade_guide#changed-xdebug.remote_enable (See: https://xdebug.org/docs/errors#CFG-C-CHANGED)

[ERROR  - 11:36:11 AM] Uncaught RuntimeException: PHP Error: Creation of dynamic property Psalm\Internal\LanguageServer\LanguageServer::$mapper is deprecated in /path/to/project/vendor/felixfbecker/advanced-json-rpc/lib/Dispatcher.php:53 in /path/to/project/vendor/vimeo/psalm/src/Psalm/Internal/ErrorHandler.php:66
Stack trace:
#0 /path/to/project/vendor/felixfbecker/advanced-json-rpc/lib/Dispatcher.php(53): Psalm\Internal\ErrorHandler::Psalm\Internal\{closure}(8192, 'Creation of dyn...', '/path/to...', 53)
#1 /path/to/project/vendor/vimeo/psalm/src/Psalm/Internal/LanguageServer/LanguageServer.php(116): AdvancedJsonRpc\Dispatcher->__construct(Object(Psalm\Internal\LanguageServer\LanguageServer), '/')
#2 /path/to/project/vendor/vimeo/psalm/src/Psalm/Internal/Analyzer/ProjectAnalyzer.php(555): Psalm\Internal\LanguageServer\LanguageServer->__construct(Object(Psalm\Internal\LanguageServer\ProtocolStreamReader), Object(Psalm\Internal\LanguageServer\ProtocolStreamWriter), Object(Psalm\Internal\Analyzer\ProjectAnalyzer))
#3 /path/to/project/vendor/vimeo/psalm/src/Psalm/Internal/Cli/LanguageServer.php(322): Psalm\Internal\Analyzer\ProjectAnalyzer->server(NULL, false)
#4 /path/to/project/vendor/vimeo/psalm/psalm-language-server(7): Psalm\Internal\Cli\LanguageServer::run(Array)
#5 {main}
(Psalm 4.30.0@d0bc6e25d89f649e4f36a534f330f8bb4643dd69 crashed due to an uncaught Throwable)

Uncaught RuntimeException: PHP Error: Creation of dynamic property Psalm\Internal\LanguageServer\LanguageServer::$mapper is deprecated in /path/to/project/vendor/felixfbecker/advanced-json-rpc/lib/Dispatcher.php:53 in /path/to/project/vendor/vimeo/psalm/src/Psalm/Internal/ErrorHandler.php:66
Stack trace:
#0 /path/to/project/vendor/felixfbecker/advanced-json-rpc/lib/Dispatcher.php(53): Psalm\Internal\ErrorHandler::Psalm\Internal\{closure}(8192, 'Creation of dyn...', '/path/to...', 53)
#1 /path/to/project/vendor/vimeo/psalm/src/Psalm/Internal/LanguageServer/LanguageServer.php(116): AdvancedJsonRpc\Dispatcher->__construct(Object(Psalm\Internal\LanguageServer\LanguageServer), '/')
#2 /path/to/project/vendor/vimeo/psalm/src/Psalm/Internal/Analyzer/ProjectAnalyzer.php(555): Psalm\Internal\LanguageServer\LanguageServer->__construct(Object(Psalm\Internal\LanguageServer\ProtocolStreamReader), Object(Psalm\Internal\LanguageServer\ProtocolStreamWriter), Object(Psalm\Internal\Analyzer\ProjectAnalyzer))
#3 /path/to/project/vendor/vimeo/psalm/src/Psalm/Internal/Cli/LanguageServer.php(322): Psalm\Internal\Analyzer\ProjectAnalyzer->server(NULL, false)
#4 /path/to/project/vendor/vimeo/psalm/psalm-language-server(7): Psalm\Internal\Cli\LanguageServer::run(Array)
#5 {main}
(Psalm 4.30.0@d0bc6e25d89f649e4f36a534f330f8bb4643dd69 crashed due to an uncaught Throwable)

[Info  - 11:36:11 AM] Connection to server got closed. Server will restart.
[INFO  - 11:36:11 AM] Psalm Version was detected as 4.30.0
[ERROR  - 11:36:12 AM] Xdebug: [Config] The setting 'xdebug.remote_autostart' has been renamed, see the upgrading guide at https://xdebug.org/docs/upgrade_guide#changed-xdebug.remote_autostart (See: https://xdebug.org/docs/errors#CFG-C-CHANGED)
Xdebug: [Config] The setting 'xdebug.remote_enable' has been renamed, see the upgrading guide at https://xdebug.org/docs/upgrade_guide#changed-xdebug.remote_enable (See: https://xdebug.org/docs/errors#CFG-C-CHANGED)

Xdebug: [Config] The setting 'xdebug.remote_autostart' has been renamed, see the upgrading guide at https://xdebug.org/docs/upgrade_guide#changed-xdebug.remote_autostart (See: https://xdebug.org/docs/errors#CFG-C-CHANGED)
Xdebug: [Config] The setting 'xdebug.remote_enable' has been renamed, see the upgrading guide at https://xdebug.org/docs/upgrade_guide#changed-xdebug.remote_enable (See: https://xdebug.org/docs/errors#CFG-C-CHANGED)

[ERROR  - 11:36:13 AM] Uncaught RuntimeException: PHP Error: Creation of dynamic property Psalm\Internal\LanguageServer\LanguageServer::$mapper is deprecated in /path/to/project/vendor/felixfbecker/advanced-json-rpc/lib/Dispatcher.php:53 in /path/to/project/vendor/vimeo/psalm/src/Psalm/Internal/ErrorHandler.php:66
Stack trace:
#0 /path/to/project/vendor/felixfbecker/advanced-json-rpc/lib/Dispatcher.php(53): Psalm\Internal\ErrorHandler::Psalm\Internal\{closure}(8192, 'Creation of dyn...', '/path/to...', 53)
#1 /path/to/project/vendor/vimeo/psalm/src/Psalm/Internal/LanguageServer/LanguageServer.php(116): AdvancedJsonRpc\Dispatcher->__construct(Object(Psalm\Internal\LanguageServer\LanguageServer), '/')
#2 /path/to/project/vendor/vimeo/psalm/src/Psalm/Internal/Analyzer/ProjectAnalyzer.php(555): Psalm\Internal\LanguageServer\LanguageServer->__construct(Object(Psalm\Internal\LanguageServer\ProtocolStreamReader), Object(Psalm\Internal\LanguageServer\ProtocolStreamWriter), Object(Psalm\Internal\Analyzer\ProjectAnalyzer))
#3 /path/to/project/vendor/vimeo/psalm/src/Psalm/Internal/Cli/LanguageServer.php(322): Psalm\Internal\Analyzer\ProjectAnalyzer->server(NULL, false)
#4 /path/to/project/vendor/vimeo/psalm/psalm-language-server(7): Psalm\Internal\Cli\LanguageServer::run(Array)
#5 {main}
(Psalm 4.30.0@d0bc6e25d89f649e4f36a534f330f8bb4643dd69 crashed due to an uncaught Throwable)

Uncaught RuntimeException: PHP Error: Creation of dynamic property Psalm\Internal\LanguageServer\LanguageServer::$mapper is deprecated in /path/to/project/vendor/felixfbecker/advanced-json-rpc/lib/Dispatcher.php:53 in /path/to/project/vendor/vimeo/psalm/src/Psalm/Internal/ErrorHandler.php:66
Stack trace:
#0 /path/to/project/vendor/felixfbecker/advanced-json-rpc/lib/Dispatcher.php(53): Psalm\Internal\ErrorHandler::Psalm\Internal\{closure}(8192, 'Creation of dyn...', '/path/to...', 53)
#1 /path/to/project/vendor/vimeo/psalm/src/Psalm/Internal/LanguageServer/LanguageServer.php(116): AdvancedJsonRpc\Dispatcher->__construct(Object(Psalm\Internal\LanguageServer\LanguageServer), '/')
#2 /path/to/project/vendor/vimeo/psalm/src/Psalm/Internal/Analyzer/ProjectAnalyzer.php(555): Psalm\Internal\LanguageServer\LanguageServer->__construct(Object(Psalm\Internal\LanguageServer\ProtocolStreamReader), Object(Psalm\Internal\LanguageServer\ProtocolStreamWriter), Object(Psalm\Internal\Analyzer\ProjectAnalyzer))
#3 /path/to/project/vendor/vimeo/psalm/src/Psalm/Internal/Cli/LanguageServer.php(322): Psalm\Internal\Analyzer\ProjectAnalyzer->server(NULL, false)
#4 /path/to/project/vendor/vimeo/psalm/psalm-language-server(7): Psalm\Internal\Cli\LanguageServer::run(Array)
#5 {main}
(Psalm 4.30.0@d0bc6e25d89f649e4f36a534f330f8bb4643dd69 crashed due to an uncaught Throwable)

[Error - 11:36:13 AM] Connection to server got closed. Server will not be restarted.
```

</p>
</details>

<details>

<summary>User Settings</summary>

<p>

```
phpExecutablePath: "/opt/homebrew/bin/php"
phpExecutableArgs: ["-dxdebug.remote_autostart=0","-dxdebug.remote_enable=0","-dxdebug_profiler_enable=0"]
psalmVersion: null
psalmScriptPath: "vendor/vimeo/psalm/psalm-language-server"
psalmScriptArgs: []
disableAutoComplete: false
maxRestartCount: 5
unusedVariableDetection: false
enableVerbose: false
connectToServerWithTcp: false
enableUseIniDefaults: false
logLevel: "INFO"
analyzedFileExtensions: [{"scheme":"file","language":"php"},{"scheme":"untitled","language":"php"}]
configPaths: ["psalm.xml","psalm.xml.dist"]
hideStatusMessageWhenRunning: true
```

</p>
</details>

**Comment by rogervila** [comment]
It does not fail with Psalm 5 and PHP 8.2

**Comment by weirdan** [comment]
Is there any reason you're not using Psalm 5?

**Comment by rogervila** [comment]
Thank you @tm1000, I am migrating multiple packages from 4 to 5, this is why I have errors with Psalm 4 and PHP 8.2.

Could the plugin show a message saying it will not run because the current Psalm version is < 5?

**Comment by tm1000** [comment]
Unfortunately the error comes from psalm not the plug-in/extension
